src/contracts/storage/Staking.ts

```diff
@@ -32,17 +35,22 @@ class StakingContract extends ContractWithTokens {
 
   private static instance: StakingContract
 
-  public stake(
+  public async stake(
     amount: string | number,
     txOptions: TxOptions,
   ): Promise<TransactionReceipt> {
-    if (amount < 0) {
-      throw new Error('amount should greater then 0')
+    if (Big(amount).lte(Big(0))) {
+      throw new CustomError('The amount to stake should be greater then 0.')
     }
 
-    const { tokenAddress } = this.getToken(txOptions.token)
+    const { from: account, token } = txOptions
+    const { tokenAddress } = this.getToken(token)
     const amountWei = convertToWeiString(amount)
 
+    await validateBalance({
+      web3: this.web3, minAmountWei: amountWei, account, token,
+    })
```

Comment:
Feel free to ignore.
```suggestion
    const minAmountWei = convertToWeiString(amount)
    const { web3 } = this

    await validateBalance({
      web3, minAmountWei, account, token,
    })
```

Reply:
Sorry for ignoring 😅 I don't like much the `const minAmountWei = convertToWeiString(amount)` as it would result in `this.methods.stake(minAmountWei, ...`